seed_intersphinx_mapping/__init__.py

```diff
@@ -127,26 +132,24 @@ def get_sphinx_doc_url(pypi_name: str) -> str:
 		Now raises :exc:`~packaging.requirements.InvalidRequirement` rather than
 		:exc:`apeye.slumber_url.exceptions.HttpNotFoundError` if the project could not be found on PyPI.
 	"""
-
-	for key, value in _get_project_links(pypi_name).items():
-
+	docs_urls = []
+	for value in _get_project_links(pypi_name):
 		# Follow redirects to get actual URL
-		r = RequestsURL(value).head(allow_redirects=True, timeout=10)
-		if r.status_code != 200:  # pragma: no cover
-			raise ValueError(f"Documentation URL not found: HTTP Status {r.status_code}.")
+		r = requests.head(value, allow_redirects=True, timeout=10)
 
-		docs_url = r.url
+		if r.status_code == 200:
+			has_extension = os.path.splitext(urlparse(r.url).path)[-1]
+			url = os.path.dirname(r.url) if has_extension else r.url
+			docs_urls.append(url)
 
-		if docs_url.endswith('/'):
-			objects_inv_url = f"{docs_url}objects.inv"
-		else:  # pragma: no cover
-			objects_inv_url = f"{docs_url}/objects.inv"
+	for docs_url in docs_urls:
+		objects_inv_url = f"{docs_url.rstrip('/')}/objects.inv"
 
 		r = requests.head(objects_inv_url)
 		if r.status_code != 200:
-			raise ValueError(f"objects.inv not found at url {objects_inv_url}: HTTP Status {r.status_code}.")
-
-		return docs_url
+			stderr_writer(f"WARNING: objects.inv not found at url {objects_inv_url}: HTTP Status {r.status_code}.")
```

Comment:
I'm reluctant to accept this change as `get_sphinx_doc_url` is a public function that other modules can use to query documentation URLs. I'd like the caller to still be able to catch the `ValueError` and display the message to the user themselves.

If you're wanting to make the warning on line 197 more specific -- to say *why* the documentation URL couldn't be obtained -- how about including the exception message text? E.g.:

```
except (ValueError, requests.exceptions.ConnectionError, requests.exceptions.Timeout) as e:
	# Couldn't get it from PyPI, trying fallback mapping
	if project_name in fallback_mapping():
		doc_url = fallback_mapping()[project_name]
		intersphinx_mapping[project_name] = (doc_url, None)
	else:
        stderr_writer(f"WARNING: Unable to determine documentation url for project {project_name}: {str(e)}")

Reply:
I understand your point, removing the `raise` was here to ensure all URLs are queried until one is good.
Due to the addition of the home-page URL in `docs_urls`, they may be more than one URL to iterate over. Keeping the `raise` makes `get_sphinx_doc_url` always return or fail on the first one, so if the documentation URL is bad, but homepage is good, it will not work.